### ISSUE TITLE
Question regarding grid normalization in CylinderTimeDataset

### ISSUE THREAD

**Issue by ZhangMingyu01** [jhagnberger/calm-pde#1]
Hello, thank you for creating and maintaining this excellent project. I've been working with the Cylinder dataset and have a question about the grid coordinate normalization in the CylinderTimeDataset class that I hope you can clarify.
In the code, I found the following lines intended to normalize the grid coordinates into a [0,1]² domain:
# Rearrange grid into [0,1]^2 domain to avoid changing the range of the model for each dataset
self.grid = self.grid / torch.tensor([1.6, 0.4])
However, when I inspected the preprocessed Cylinder dataset (train_grid.npy), I observed that the actual grid coordinates have a value range of approximately [0, 1000], not [0, 1.6] or [0, 0.4].
This leads to my question: With the current divisors [1.6, 0.4], the normalized grid would actually end up in the range of approximately [0, 625] for the x-axis and [0, 2500] for the y-axis, which contradicts the goal stated in the comment of mapping to a [0,1]^2 domain.
Would it be more appropriate to normalize using the actual data range, for example:
self.grid = self.grid / 1000.0
I wanted to check if:
1、The use of [1.6, 0.4] is intentional and based on a specific physical scaling or historical reason I might have missed.
2、Or if this might be an oversight that should be corrected to ensure proper normalization.
Thank you very much for your time and guidance. I appreciate your help in clarifying this.

**Comment by jhagnberger** [comment]
Thank you for using our project and for your question.

The cylinder dataset has a spatial domain of [0, 1.6] and [0, 0.4], thus, the scaling factors of [1.6, 0.4] are correct. However, the number of grid points in the cylinder dataset varies for each sample (e.g., one sample could have 1800 points and another 1900 points).

To store all samples in one array and to train the models with a batch size larger than 1, we have to apply padding. The physical field data is padded with zeros, and the positions with 1000 in each dimension (see https://github.com/jhagnberger/calm-pde/blob/446fb46c74931c125e810d32dd706288b766b995/calm_pde/data/datasets.py#L626). Using 1000 as the padding position has the advantage that the CALM-PDE encoder will directly ignore the padding points because these points are out of scope. Since this trick does not work for all models, the dataset also provides a mask that other models can use to ensure proper masking of the padding points.

Unfortunately, we did not document this procedure. Thus, we will add a description to the Hugging Face dataset to provide clarity.

**Comment by ZhangMingyu01** [comment]
Thank you very much for your answer. I understand the issue now.
